judoscale-ruby/lib/judoscale/utilization_middleware.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require "singleton"
+require "judoscale/utilization_tracker"
+
+module Judoscale
+  class UtilizationMiddleware
```

Comment:
If we're going to make it enabled by default and maybe not even need to allow disabling, I wonder if we really need a separate middleware, or if we should bundle it all together with the `RequestMiddleware`, and just wrap it with the tracker start/incr/decr code.

I mean, the separation is nice, but I couldn't always get the same separation on Python especially considering some of the client code is responsible to add the middleware (much like a Rack app does) so they'd have to change their app code to start tracking utilization, so maybe it'd make sense to have it all just handled by our single middleware?